graphdb/janus/src/main/java/org/apache/atlas/repository/graphdb/janus/AtlasJanusGraph.java

```diff
@@ -330,13 +332,16 @@ public void exportToGson(OutputStream os) throws IOException {
 
     @Override
     public GremlinGroovyScriptEngine getGremlinScriptEngine() {
+        return scriptEngine;
```

Comment:
GremlinGroovyScriptEngine returned here may not safe for use in multiple threads simultaneously i.e. side-effects of an execution in a thread might be visible in other threads. I suggest to use a ThreadLocal<GremlinGroovyScriptEngine>, to instantiate and use one script engine per thread.

Reply:
according to tinkerpop github code,the GremlinGroovyScriptEngine class is thread-safe.below is the link of tinkerpop test code for thread-safe feature ( shouldBeThreadSafe() method ):
https://github.com/apache/tinkerpop/blob/master/gremlin-groovy/src/test/java/org/apache/tinkerpop/gremlin/groovy/jsr223/GremlinGroovyScriptEngineOverGraphTest.java
thanks.